addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -280,6 +283,60 @@ export class SelfOrder extends Reactive {
         );
 
         this.currentCategory = this.pos_category.length > 0 ? [...this.categoryList][0] : null;
+
+        if (this.kitchen_printers) {
+            for (const printerConfig of Object.values(this.kitchen_printers)) {
+                const printer = this.create_printer(printerConfig);
+                if (printer) {
+                    printer.config = printerConfig;
+                    this.kitchenPrinters.push(printer);
+                }
+            }
+        }
```

Comment:
```suggestion
        for (const printerConfig of Object.values(this.kitchen_printers)) {
            const printer = this.create_printer(printerConfig);
            if (printer) {
                printer.config = printerConfig;
                this.kitchenPrinters.push(printer);
            }
        }
```

No need for the `if` since `kitchen_printers` is always an object (even if it can be an empty object).